staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -472,80 +478,65 @@ type processorListener struct {
 
 func newProcessListener(handler ResourceEventHandler, requestedResyncPeriod, resyncPeriod time.Duration, now time.Time) *processorListener {
 	ret := &processorListener{
-		pendingNotifications:  []interface{}{},
 		nextCh:                make(chan interface{}),
+		addCh:                 make(chan interface{}),
```

Comment:
I completely agree that the benchmark is not perfect. I'll happily implement something in addition to it but nothing comes to my mind. If you have any ideas, please share :)

I don't think in practice it matters though - events are coming via the network so delays in order of 0.1-10ms should probably be expected. And we are talking about nanoseconds here.